### PR TITLE
add feature: control filters by group

### PR DESCRIPTION
**Pull request by JeanTracker**
It is inconvenient to control filters one by one. This feature is added to group filters according to their purpose and control them collectively.

### CODE REVIEW

**Comment by JeanTracker** [comment]
This is a PR for first feature of #7 .

**Comment by JeanTracker** [comment]
The README file will be updated after all features, including feature #7, have been fully implemented.

File: src/filterTreeViewProvider.ts [hunk 1]
```diff
@@ -1,68 +1,103 @@
...
     return element;
   }
 
-  //getChildren(filterItem) returns empty list because filters have no children.
+  //getChildren(vscode.TreeItem) returns empty list because filters have no children.
   //getChildren() returns the root elements (all the filters)
-  getChildren(element?: FilterItem): FilterItem[] {
-    if (element) {
-      return [];
+  getChildren(element?: vscode.TreeItem): Thenable<vscode.TreeItem[]> {
```
Comment (SoySauceFor3) [inline src/filterTreeViewProvider.ts:14]: Why do you choose to use Thenable instead of just returning a vscode.TreeItem[]?

**Review by SoySauceFor3 — CHANGES_REQUESTED** [review]

File: src/filterTreeViewProvider.ts (hunk 1 above)
Comment (JeanTracker) [reply src/filterTreeViewProvider.ts:14]: I have confirmed that in various official examples [vscode-extension-samples](https://github.com/microsoft/vscode-extension-samples), getChildren is implemented using Thenable and handled asynchronously.

File: src/focusProvider.ts [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 //VSCode uses this provider to generate virtual read-only files based on real files
 export class FocusProvider implements vscode.TextDocumentContentProvider {
-  filterArr: Filter[];
+  groupArr: Group[];
 
-  constructor(filterArr: Filter[]) {
-    this.filterArr = filterArr;
+  constructor(groupArr: Group[]) {
+    this.groupArr = groupArr;
   }
```
[the remark this answers is not in the document]
Comment (JeanTracker) [reply src/focusProvider.ts:13]: Yes, you're right but I just fixed filter to group :)
Could you fix this later?

File: src/commands.ts [hunk 3]
```diff
@@ -93,29 +100,33 @@ export function importFilters(state: State) {
...
-      const array = parsed as any[];
-      array.forEach((filterText) => {
-        if (
-          typeof filterText.regexText === "string" &&
-          typeof filterText.color === "string" &&
-          typeof filterText.isHighlighted === "boolean" &&
-          typeof filterText.isShown === "boolean"
-        ) {
-          const id = `${Math.random()}`;
+      parsed.groups.map((g: Group) => {
```
[the remark this answers is not in the document]
Comment (JeanTracker) [reply src/commands.ts]: You're right. I fixed it :)

File: src/filterTreeViewProvider.ts (hunk 1 above)
Comment (SoySauceFor3) [reply src/filterTreeViewProvider.ts:14]: yes but the return type can also be non async, and there is no clear reason why needing the async here. As far as I know Promise.resolve() creates a async that is in a sync underneath. 
I don't have a very strong opinion here, so the above is just for discussion here.

**Review by SoySauceFor3 — APPROVED** [review]
Thanks for resolving my comments.

**Pull request merged**